Simplenote/src/main/java/com/automattic/simplenote/ThemedAppCompatActivity.java

```diff
@@ -12,10 +16,49 @@
  * Abstract class to apply theme based on {@link PrefUtils#PREF_STYLE_INDEX}
  * to any {@link AppCompatActivity} that extends it.
  */
-abstract public class ThemedAppCompatActivity extends AppCompatActivity {
+abstract public class ThemedAppCompatActivity extends AppCompatActivity implements SharedPreferences.OnSharedPreferenceChangeListener {
+    private Boolean mThemeChanged = false;
+
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
+        ThemeUtils.setTheme(this);
         super.onCreate(savedInstanceState);
-        setTheme(ThemeUtils.getStyle(ThemedAppCompatActivity.this));
+        setTheme(ThemeUtils.getStyle(this));
+        PreferenceManager.getDefaultSharedPreferences(this).registerOnSharedPreferenceChangeListener(this);
+    }
+
+    @Override
+    protected void onRestart() {
+        super.onRestart();
+        if(mThemeChanged) {
+            recreate();
+            mThemeChanged = false;
+        }
+    }
+
+    @Override
+    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
+        if(key.equals(PrefUtils.PREF_THEME) || key.equals(PrefUtils.PREF_STYLE_INDEX)) {
+            if(getLifecycle().getCurrentState().isAtLeast(Lifecycle.State.RESUMED)) {
```

Comment:
```suggestion
        if (mThemeChanged) {
            recreate();
            mThemeChanged = false;
        }
    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
        if (key.equals(PrefUtils.PREF_THEME) || key.equals(PrefUtils.PREF_STYLE_INDEX)) {
            if (getLifecycle().getCurrentState().isAtLeast(Lifecycle.State.RESUMED)) {
```

There is a space missing between the `if` and the `(` in these three statements.